analyses/cell-type-neuroblastoma-04/run-analysis.sh

```diff
@@ -18,16 +28,39 @@ module_dir=$(dirname "${BASH_SOURCE[0]}")
 cd ${module_dir}
 
 # Define and create directories
+# do this before flag setup since some directory names are used
 data_dir="../../data/current/SCPCP000004"
 script_dir="scripts"
 ref_dir="references"
 scratch_dir="scratch"
+results_dir="results"
+singler_results_dir="${results_dir}/singler"
 mkdir -p $ref_dir
 mkdir -p $scratch_dir
+mkdir -p $singler_results_dir
+
+###################################################################
+######################## Set up variables #########################
+###################################################################
+
+threads=${threads:-4} # default 4 threads
+aggregate_singler=${aggregate_singler:-1} # default perform aggregation
+
+# Set up the singler aggregation accordingly
+if [[ $aggregate_singler -eq 1 ]]; then
+    aggregate_flag="--aggregate_reference"
+    singler_model_file="${scratch_dir}/singler-model_nbatlas_aggregated.rds"
+    singler_aggregate_type="aggregated" # sub-directory where results will be saved
+else
+    aggregate_flag=""
+    singler_model_file="${scratch_dir}/singler-model_nbatlas_not-aggregated.rds"
+    singler_aggregate_type="not-aggregated"
+fi
```

Comment:
```suggestion
if [[ $aggregate_singler -eq 1 ]]; then
    aggregate_flag="--aggregate_reference"
    singler_aggregate_type="aggregated" # sub-directory where results will be saved
else
    aggregate_flag=""
    singler_aggregate_type="not-aggregated"
fi
singler_model_file="${scratch_dir}/singler-model_nbatlas_${singler_aggregate_type}.rds"
```
You can pull this out and use the aggregate type variable to define the file.